csm-1.4/CASMINST-6896-gpg-keys/install-hotfix.sh

```diff
@@ -0,0 +1,256 @@
+    echo >&2 "No NCNs found in /etc/hosts! This NCN is not initialized, /etc/hosts should have content."
+    exit 1
+  fi
+fi
+
+export NCNS=()
+for ncn in "${EXPECTED_NCNS[@]}"; do
+  if ping -c 1 "$ncn" >/dev/null 2>&1; then
+    NCNS+=("$ncn")
+  else
+    echo >&2 "Failed to ping [$ncn]; skipping hotfix for [$ncn]"
+  fi
+done
+
+function patch_running_ncns {
+  local gpg_key_file_name="${GPG_KEY_FILE_NAME}"
+  local PDSH_SSH_ARGS_APPEND="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+  local SSH_ARGS="${PDSH_SSH_ARGS_APPEND}"
+
+  for ncn in "${NCNS[@]}"; do
+    scp ${SSH_ARGS} "${ROOT_DIR}/keys/${gpg_key_file_name}" "${ncn}:/tmp/${gpg_key_file_name}"
+  done
+
+  pdsh -S -b -w "$(printf '%s,' "${NCNS[@]}")" '
+    rpm --import /tmp/'"${gpg_key_file_name}"'
+    rm -vf /tmp/'"${gpg_key_file_name}"'
+  ' || { echo >&2 'Failed to import the new HPE GPG key on one or more nodes!'; return 1; }
+}
+
+patch_running_ncns || { echo >&2 'Failed to import the key on one or more running NCNs!'; exit 1; }
+
+if [ "$running_system_only" -eq 1 ]; then
+  cat >&2 <<EOF
++ Hotfix installed
+${0##*/}: OK
+EOF
+  exit 0
+fi
+
+############################################### CFS ########################################################
+
+workdir="$(mktemp -d)"
+[ -z "${DEBUG:-}" ] && trap 'rm -fr '"${workdir}"'' ERR INT EXIT RETURN || echo "DEBUG was set in environment, $workdir will not be cleaned up."
+
+# Set credentials for the VCS
+PW=$(kubectl -n services get secret vcs-user-credentials -o jsonpath='{.data.vcs_password}' | base64 -d)
+
+# Fetch the latest commit from the specified branch
+CSM_CONFIG_COMMIT=$(git ls-remote "https://crayvcs:${PW}@api-gw-service-nmn.local/vcs/cray/csm-config-management.git" "refs/heads/cray/csm/${CSM_CONFIG_VERSION}" | awk '{print $1}')
+unset PW
+
+[ -z "$CSM_CONFIG_COMMIT" ] && { echo >&2 "Failed to retrieve the latest commit from csm-config branch cray/csm/${CSM_CONFIG_VERSION}. Aborting."; exit 1; }
+
+# Update the kubernetes secret with our new GPG key if the new key isn't present
+NEW_KEY_PATH="${ROOT_DIR}/keys/${GPG_KEY_FILE_NAME}"
+NEW_KEY_SIGNATURE="$(gpg --list-packets "${NEW_KEY_PATH}")"
+NEW_KEY_ENCODED="$(base64 -w 0 "${NEW_KEY_PATH}")"
+mapfile -t EXISTING_K8S_KEYS < <(kubectl -n services get secret hpe-signing-key -o jsonpath='{.data}' | jq -r 'keys[]')
+
+KEY_PRESENT=0
+for key in "${EXISTING_K8S_KEYS[@]}"; do
+  EXISTING_KEY_SIGNATURE="$(kubectl -n services get secret hpe-signing-key -o jsonpath="{.data.${key/./\\.}}" | base64 -d | gpg --list-packets)"
+
+  if [ "${EXISTING_KEY_SIGNATURE}" = "${NEW_KEY_SIGNATURE}" ]; then
+    echo "Key ${key} is already present in the secret. Refusing to add it again."
+    KEY_PRESENT=1
+    break
+  fi
+done
+
+if [ "${KEY_PRESENT}" -eq 0 ]; then
+  echo "Key ${key} is not present, adding it to the secret: hpe-signing-key"
+  kubectl patch secret hpe-signing-key -n services -p="{\"data\":{\"${GPG_KEY_FILE_NAME}\":\"${NEW_KEY_ENCODED}\"}}"
+fi
```

Comment:
```suggestion
  kubectl patch secret hpe-signing-key -n services -p="{\"data\":{\"${GPG_KEY_FILE_NAME}\":\"${NEW_KEY_ENCODED}\"}}"
else
    echo "Key ${key} was already present in Kubernetes secret: ${kubernetes_secret}"
fi
```

Add a global var for hpe-signing-key:
```bash
KUBERNETES_SECRET="${KUBERNETES_SECRET:-hpe-signing-key}"
```

Reply:
If you're okay with this, let's copy it into the other script too.